agent/consul/rpc.go

```diff
@@ -789,17 +796,19 @@ func (s *Server) raftApplyWithEncoder(t structs.MessageType, msg interface{}, en
 			// apply function. Downstream client code expects to see any error
 			// from the FSM (as opposed to the apply itself) and decide whether
 			// it can retry in the future's response.
-			return ErrChunkingResubmit, nil
+			return nil, ErrChunkingResubmit
```

Comment:
Hmm, this `return nil, ErrChunkingResubmit` part of the change I need to look at. The handling of this in `canRetry` is strange.

Reply:
Ya, the handling of `ErrChunkingResubmit` seems completely broken. It's only ever referenced in one place here: https://github.com/hashicorp/consul/blob/master/agent/consul/rpc.go#L530-L539

But looking at the callers to `canRetry`, there's no way that an error is ever going to be passed as the `args` argument:

* https://github.com/hashicorp/consul/blob/a02245b75af59b8af8545639c79f2dd571b0e174/agent/consul/client.go#L289 - args is going to be the `Request` struct for the RPC
* https://github.com/hashicorp/consul/blob/a02245b75af59b8af8545639c79f2dd571b0e174/agent/consul/rpc.go#L607 - `info` is of type `RPCInfo`, not an error.


So it looks like this was always broken.

Reply:
I pushed another commit which should fix this problem.